src/routes/tools/extreme-precipitation/_SettingsPanel.svelte

```diff
@@ -0,0 +1,217 @@
+<script>
+  import { createEventDispatcher } from "svelte";
+  import { NumberInput } from "carbon-components-svelte";
+
+  import {
+    PRIORITY_10_MODELS,
+    DEFAULT_SCENARIOS,
+    SELECT_LOCATION_DESCRIPTION,
+  } from "../_common/constants";
+  import {
+    THRESHOLD_TYPES,
+    RETURN_PERIODS,
+    POLYGON_AGGREGATE_FUNCTIONS,
+  } from "./_constants";
+
+  import {
+    SelectClimvar,
+    SelectScenario,
+    RadioBtnGroup,
+    SelectModels,
+    Select,
+  } from "~/components/tools/Settings";
+  import { LearnMoreButton } from "~/components/tools/Partials";
+  import { StaticMap } from "~/components/tools/Location";
+
+  import { locationStore, scenarioStore, modelsStore } from "../_common/stores";
+  import {
+    indicatorList,
+    indicatorStore,
+    durationStore,
+    thresholdStore,
+    thresholdTypeStore,
+    returnPeriodStore,
+    aggregateFnStore,
+  } from "./_store";
+
+  export let learnMoreContent;
+
+  const {
+    durationInfo,
+    indicatorInfo,
+    periodInfo,
+    thresholdTypeInfo,
+    aggregationInfo,
+  } = learnMoreContent;
+  const dispatch = createEventDispatcher();
+  const { location, boundary } = locationStore;
+  const { indicator } = indicatorStore;
+
+  $: isPolygonBoundary = $boundary.id !== "locagrid";
+
+  function showLearnMore({ slugs = [], content = "", header = "Glossary" }) {
+    dispatch("showLearnMore", { slugs, content, header });
+  }
+
+  function showChangeLocation() {
+    dispatch("showLoadLocation");
+  }
+
+  function changeScenario(e) {
+    scenarioStore.set(e.detail.id);
+  }
+
+  function changeIndicator(e) {
+    indicatorStore.set(e.detail.id);
+  }
+
+  function changeDuration(e) {
+    durationStore.set(+e.detail);
+  }
+
+  function changeModels(e) {
+    modelsStore.set(e.detail.selectedIds);
+  }
+
+  function changeThreshold(e) {
+    thresholdStore.set(e.detail);
+  }
+
+  function changeThresholdType(e) {
+    thresholdTypeStore.set(e.detail.id);
+  }
+
+  /**
+   * The Return Period Select component is displayed only for the Intensity Indicator.
+   * The component is re-mounted after users browser other indicators and come back to Intensity.
+   * Parsing the e.detail value as integer ensures that the Return Period last selected by the user
+   * is displayed correctly on re-mount
+   **/
+  function changeReturnPeriod(e) {
+    returnPeriodStore.set(parseInt(e.detail));
```

Comment:
when using `parseInt` it's a good practice to pass the arg for `radix`, otherwise unintentional things can happen if the value isn't a number; see: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt

Reply:
thanks. didn't know about `radix`. implemented in https://github.com/berkeley-gif/caladapt-website-2021/pull/211/commits/f12739c37499ee72aa57363e6b759e38f017f83c